generate.py

```diff
@@ -42,6 +44,10 @@ def generate(
     empty[:T] = idx
     idx = empty
 
+    if use_xla:
```

Comment:
Should we just check `idx.device.type` so that we don't need to pass an argument around?